File: benchmarks/number_analyzer.py
import math

def count_digits_recursive(n: int) -> int:
    n = abs(n)
    if n <= 9:
        return 1
    return 1 + count_digits_recursive(n // 10)

def count_digits_iterative(n: int) -> int:
    if n == 0:
        return 1
    n = abs(n)
    count = 0
    while n != 0:
        n //= 10
        count += 1
    return count

def analyze_number(n: int) -> dict[str, int]:
    return {
        "original_number": n,
        "digit_count_recursive": count_digits_recursive(n),
        "digit_count_iterative": count_digits_iterative(n),
        "is_negative": 1 if n < 0 else 0,
        "is_even": 1 if n % 2 == 0 else 0
    }

def test_analyze_positive():
    res = analyze_number(156)
    assert res["original_number"] == 156
    assert res["digit_count_recursive"] == 3
    assert res["digit_count_iterative"] == 3
    assert res["is_negative"] == 0
    assert res["is_even"] == 1

def test_analyze_negative():
    res = analyze_number(-7)
    assert res["original_number"] == -7
    assert res["digit_count_recursive"] == 1
    assert res["digit_count_iterative"] == 1
    assert res["is_negative"] == 1
    assert res["is_even"] == 0

if __name__ == "__main__":
    test_analyze_positive()
    test_analyze_negative()

File: benchmarks/NumberAnalyzer.java
import java.util.HashMap;
import java.util.Map;

public class NumberAnalyzer {
    public static int countDigitsRecursive(int n) {
        int num = Math.abs(n);
        if (num <= 9) return 1;
        return 1 + countDigitsRecursive(num / 10);
    }
    
    public static int countDigitsIterative(int n) {
        if (n == 0) return 1;
        int num = Math.abs(n);
        int count = 0;
        while (num != 0) {
            num = num / 10;
            count++;
        }
        return count;
    }
    
    public static Map<String, Integer> analyzeNumber(int n) {
        Map<String, Integer> res = new HashMap<>();
        res.put("original_number", n);
        res.put("digit_count_recursive", countDigitsRecursive(n));
        res.put("digit_count_iterative", countDigitsIterative(n));
        res.put("is_negative", n < 0 ? 1 : 0);
        res.put("is_even", n % 2 == 0 ? 1 : 0);
        return res;
    }

    public static void testAnalyzePositive() {
        Map<String, Integer> res = analyzeNumber(156);
        assert res.get("original_number") == 156;
        assert res.get("digit_count_recursive") == 3;
        assert res.get("digit_count_iterative") == 3;
        assert res.get("is_negative") == 0;
        assert res.get("is_even") == 1;
    }

    public static void testAnalyzeNegative() {
        Map<String, Integer> res = analyzeNumber(-7);
        assert res.get("original_number") == -7;
        assert res.get("digit_count_recursive") == 1;
        assert res.get("digit_count_iterative") == 1;
        assert res.get("is_negative") == 1;
        assert res.get("is_even") == 0;
    }

    public static void main(String[] args) {
        // Run tests
        testAnalyzePositive();
        testAnalyzeNegative();
    }
}
